src/routes/Attendance.ts

```diff
@@ -0,0 +1,299 @@
+import StatusCodes from "http-status-codes";
+import e, { Request, Response } from "express";
+import { join } from "path";
+import fs from "fs";
+import logger from "@shared/Logger";
+import path from "path/posix";
+import util from "util";
+
+import { FieldSet, Records } from "airtable";
+
+import StudentDao from "@daos/Airtable/StudentDao";
+import CanvasCoursesDao from "@daos/Airtable/CanvasCoursesDao";
+import ModulesDao from "@daos/Canvas/ModulesDao";
+import { paramMissingError } from "@shared/constants";
+import SubmissionDao from "@daos/Canvas/SubmissionDao";
+import Module from "@entities/Module";
+import Submission from "@entities/Submission";
+import ModuleItem from "@entities/ModuleItem";
+
+const studentDao = new StudentDao();
+const canvasCoursesDao = new CanvasCoursesDao();
+const modulesDao = new ModulesDao();
+const submissionDao = new SubmissionDao();
+
+const { BAD_REQUEST, CREATED, OK } = StatusCodes;
+
+interface IMilestone {
+  name: string;
+  itemId: number;
+  points: number;
+  completed: boolean;
+}
+
+/**
+ * Get a set of learners' student records from Airtable by their email.
+ *
+ * @param learners
+ * @returns
+ */
+async function getStudentRecords(
+  learners: Record<string, unknown>[]
+): Promise<Records<FieldSet>> {
+  const emails: string[] = learners.map((learner) => learner.Email as string);
+  return await studentDao.getByEmails(emails);
+}
+
+/**
+ * Merge each learner's Lambda ID and Labs Role from their student record.
+ * (Mutates learners array.)
+ *
+ * @param learners
+ * @returns
+ */
+function mergeStudentRecords(
+  learners: Record<string, unknown>[],
+  studentRecords: Records<FieldSet>
+): Record<string, unknown>[] {
+  for (const learner of learners) {
+    const airtableRecord = studentRecords.find(
+      (elm) => elm.fields["Email"] == learner["Email"]
+    );
+    if (!airtableRecord) {
+      learner["Lambda ID"] = null;
+      learner["Labs Role"] = null;
+    } else {
+      learner["Lambda ID"] = airtableRecord.fields["Lambda ID"];
+      learner["Labs Role"] = airtableRecord.fields["Labs Role"];
+    }
+  }
+
+  return learners;
+}
+
+/**
+ * Look up a learner's Objectives course ID in Airtable by their Labs role.
+ *
+ * @param labsRole
+ * @returns
+ */
+async function getObjectivesCourseId(labsRole: string): Promise<number | null> {
```

Comment:
This should be returning just `number | null` and not a promise.

Reply:
Never mind on these. I had a lint error that I thought was for the exact same signature but it was an issue with my return statement.